modules/tensor_mechanics/include/userobjects/NodalPatchRecoveryBase.h

```diff
@@ -29,6 +29,16 @@ class NodalPatchRecoveryBase : public ElementUserObject
    */
   virtual Real nodalPatchRecovery(const Point & p, const std::vector<dof_id_type> & elem_ids) const;
 
+  /**
+   * Compares whether a calling object has the same block restrictio as this user object. This is
+   * handy to make sure there is consistency between nodal patch recovery methods in the auxiliary
+   * kernels and the user objects.
+   *
+   * @param aux_kernel_subdomains Set of subdomains on which the external object acts upon.
+   * @return Whether the extenal object and this object act on the same subdomains.
+   */
+  bool isBlockRestrictionSame(const std::set<SubdomainID> & aux_kernel_subdomains) const;
```

Comment:
> Is this too restrictive? For example, what happens if the userobject is restricted A, B, C, while the auxkernel is restricted to just A, should we error in that case?

Yeah, we can consider this way too restrictive. I was thinking that both aux kernel and user object should match since the purpose is to recover the properties in all the selected blocks. But will change that.